src/vm/mlinfo.cpp

```diff
@@ -1946,7 +1972,11 @@ MarshalInfo::MarshalInfo(Module* pModule,
                 m_resID = IDS_EE_BADMARSHAL_PTR;
                 IfFailGoto(E_FAIL, lFail);
             }
-            m_type = ( (sizeof(void*)==4) ? MARSHAL_TYPE_GENERIC_4 : MARSHAL_TYPE_GENERIC_8 );
+#ifdef _TARGET_64BIT_
+            m_type = MARSHAL_TYPE_GENERIC_8;
+#else
+            m_type = MARSHAL_TYPE_GENERIC_4;
+#endif
```

Comment:
These are technical identical since `(sizeof(void*)==4)` is a constant. Please put this back.

Reply:
Throughout.

---
In reply to: [338220905](https://github.com/dotnet/coreclr/pull/26340#discussion_r338220905) [](ancestors = 338220905)

Reply:
These aren't identical in the case of crossgening for Linux arm32 from a Linux x64 machine using the cross-bitness build of crossgen. Since this code is used when jitting IL stubs with native structure types, we need to make sure the type here matches type on the target architecture, not the host.